media_service/models.py

```diff
@@ -164,9 +168,11 @@ class Meta:
     def save(self, *args, **kwargs):
         if not self.sort_order:
             self.sort_order = self.next_sort_order({"course__pk": self.course.pk})
+        if not (isinstance(self.metadata, basestring) and len(self.metadata) > 0 and '[]' == self.metadata[0] + self.metadata[-1]):
```

Comment:
That approach is much more robust and readable IMO so I'm going to go ahead and make that change.  No sense doing string manipulation when we can just parse the darn thing.